spec/routines/create_user_spec.rb

```diff
@@ -2,37 +2,71 @@
 
 describe CreateUser do
 
-  context "when ensure_no_errors is false" do
-
-    it "fails with an error when the username is taken" do
-      FactoryGirl.create(:user, username: "bubba")
-      outcome = nil
-      expect {
-        outcome = CreateUser.call(username: "bubba", ensure_no_errors: false, state: 'activated')
-      }.to change{User.count}.by 0
-      expect(outcome.errors.has_offending_input?(:username)).to be_truthy
+  context "when all validations pass" do
+    it "creates a new user" do
+      expect{
+        CreateUser.call(username: "unclebob", title: "Mr.", first_name: "Robert", last_name: "Martin", state: "activated")
+      }.to change{User.count}.by 1
     end
-
   end
 
-  context "when ensure_no_errors is true" do
-    it "succeeds when a username is already taken" do
-      FactoryGirl.create(:user, username: "bubba")
-      outcome = nil
-      expect {
-        outcome = CreateUser.call(username: "bubba", ensure_no_errors: true, state: 'activated')
-      }.to change{User.count}.by 1
-      expect(outcome.errors).to be_empty
+  context "when validations fail" do
+    context "when ensure_no_errors is false" do
+      it "returns errors on invalid input" do
```

Comment:
I think this spec could/should be split into two - one for the exception check, and one for the errors check.